### PR TITLE
Kafka channel reconciler tests - except auth cases

### PR DESCRIPTION
**Pull request by aliok**
Fixes #

## Proposed Changes

- Going to have all Kafka channel reconciler tests in this PR
- I will do the tests with auth later, once we decide on what to do about the configmap
- Let's keep the TODOs

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

**Review by pierDipi — APPROVED** [review]
As TODOs, I'd add, tests where there are no CM updates.

/lgtm

File: control-plane/pkg/reconciler/base/reconciler.go [hunk 1]
```diff
@@ -173,8 +173,15 @@ func GetDataPlaneConfigMapData(logger *zap.Logger, dataPlaneConfigMap *corev1.Co
 
 		logger.Warn("Failed to unmarshal config map", zap.Error(err))
 
+		// Because of https://github.com/golang/protobuf/issues/1314, we cannot use the error message in the unit tests.
+		// Also, the error here should only happen when there is an implementation issue as the configmap is only
+		// to be touched by the controller.
+		// Thus, let's not have the error message in the returned error. Instead, let's just print it.
+
+		logger.Error(fmt.Sprintf("failed to unmarshal contract: '%s' - %s", dataPlaneDataRaw, err))
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/base/reconciler.go:181]:
```suggestion
		logger.Error("Failed to unmarshal contract", zap.Any("content", dataPlaneDataRaw) , zap.Error(err))
```

File: control-plane/pkg/reconciler/base/reconciler.go [hunk 2]
```diff
@@ -173,8 +173,15 @@ func GetDataPlaneConfigMapData(logger *zap.Logger, dataPlaneConfigMap *corev1.Co
 
 		logger.Warn("Failed to unmarshal config map", zap.Error(err))
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/base/reconciler.go:174]: Can we can also remove this warning?

Comment (aliok) [reply control-plane/pkg/reconciler/base/reconciler.go:174]: Done

**Review by pierDipi — APPROVED** [review]
/lgtm

**Pull request merged**